src/persist/src/persist.proto

```diff
@@ -0,0 +1,76 @@
+// Copyright Materialize, Inc. and contributors. All rights reserved.
+//
+// Use of this software is governed by the Business Source License
+// included in the LICENSE file.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0.
+
+// See https://developers.google.com/protocol-buffers for what's going on here.
+
+syntax = "proto3";
+
+package gen;
+
+message ProtoMeta {
+    uint64 seqno = 2;
+    repeated ProtoStreamRegistration id_mapping = 3;
```

Comment:
just for my own knowledge: whats the runtime/storage cost of using protobuf map types? is that a viable choice for these fields?

Reply:
from https://developers.google.com/protocol-buffers/docs/overview#backwards_compatibility

a map field is the same as a repeated message with two fields (key and value). so e.g. if this was just (int64 id) -> (string name), they'd be equivalent on the wire and at runtime would be whatever the difference is between a hashmap and a vec of 2-tuples. since we also need to group the codec names in, a proto map would have a tiny bit extra of extra overhead, since the encoding would be something like `[message { key: int64, message { name, key_codec, val_codec} }]` (note the extra message). but (for better or worse) proto encoding uses varints so the overhead of this extra message in practice will be 2 bytes per entry (1 byte for field id + field type, 1 byte for message encoded length)

honestly I just didn't remember that proto maps were a thing when I did this. do you think that's worth the extra overhead?

Reply:
I think its worth it iff it means we can get rid of all of the annoying "this is a vector but really its meant to hold a set, and we need all of these complicated validations to guarantee key-uniqueness" because i find those to be the biggest pain point of keeping these as vectors. otherwise, i don't have a strong feeling

Reply:
okay, I think this all depends on how the validation stuff shakes out and whether we keep stuff like ArrangementMeta around or switch to directly constructing things from protos. we'll have more breaking changes before this settles down (plus it's possible to do without a breaking change), so left it as a TODO

Reply:
haha jk. convinced myself we should do this now

Reply:
Done